sklearn/metrics/cluster/_unsupervised.py

```diff
@@ -174,7 +187,7 @@ def silhouette_samples(X, labels, *, metric="euclidean", **kwds):
 
     Parameters
     ----------
-    X : array-like of shape (n_samples_a, n_samples_a) if metric == \
+    X : {array-like, sparse matrix} of shape (n_samples_a, n_samples_a) if metric == \
```

Comment:
There is something quite difficult to tract in the code here. We accept both CSC and CSR matrix but the reduce function only works on CSR matrix. Looking at the code, the `pairwise_distance` will make the conversion from CSC to CSR but I think it would be better to only accept CSR implying that we convert `X` to CSR when other formats at the first `check_array`.

It is more explicit.

Reply:
Yes, that's a good point. I will update the code to only accept CSR sparse matrices.